Cubical/HITs/Ints/QuoInt/Properties.agda

```diff
@@ -0,0 +1,67 @@
+{-# OPTIONS --cubical --safe #-}
+module Cubical.HITs.Ints.QuoInt.Properties where
+
+open import Cubical.Core.Everything
+open import Cubical.Foundations.Prelude
+
+open import Cubical.HITs.Ints.QuoInt.Base
+  renaming (_+ℤ_ to _+_)
+open import Cubical.Data.Nat
+  hiding (+-comm; +-zero; _+_)
+
++-pos-0 : ∀ a → pos 0 + a ≡ a
++-pos-0 (pos zero) = refl
++-pos-0 (pos (suc n)) = cong sucℤ (+-pos-0 (pos n))
++-pos-0 (neg zero) = posneg
++-pos-0 (neg (suc n)) = cong predℤ (+-pos-0 (neg n))
++-pos-0 (posneg i) j = posneg (i ∧ j)
+
++-neg-0 : ∀ a → neg 0 + a ≡ a
++-neg-0 (neg zero) = refl
++-neg-0 (neg (suc n)) = cong predℤ (+-neg-0 (neg n))
++-neg-0 (pos zero) = sym posneg
++-neg-0 (pos (suc n)) = cong sucℤ (+-neg-0 (pos n))
++-neg-0 (posneg i) j = posneg (i ∨ ~ j)
+
++-pos-suc : ∀ a b → sucℤ (pos b + a) ≡ (pos (suc b) + a)
++-pos-suc (pos zero) b = refl
++-pos-suc (pos (suc n)) b = cong sucℤ (+-pos-suc (pos n) b)
++-pos-suc (neg zero) b = refl
++-pos-suc (posneg i) b = refl
++-pos-suc (neg (suc n)) b = simpleHcomp
+  (sym (sucPredℤ (pos b + neg n)))
+  (sym (predSucℤ (pos b + neg n)))
+  (cong predℤ (+-pos-suc (neg n) b))
+  -- the following hcomp does not termination-check:
+  -- hcomp (λ j → λ
+  --  { (i = i0) → sucPredℤ (pos b + neg n) (~ j)
+  --  ; (i = i1) → {!predℤ ((+-pos-suc (neg n) b) j)!}
+  --  }) (predSucℤ (pos b + neg n) (~ i))
```

Comment:
I suggest that Agda can do some enhancements here.

Reply:
This seems like a bug to me... @Saizan ?

Reply:
This is most likely an instance of https://github.com/agda/agda/issues/2100 , as pattern matching lambdas are essentially desugared to definitions in a where block.

Reply:
Well, it's a pity..